tests/rkt_socket_activation_test.go

```
@@ -0,0 +1,156 @@
+	"testing"
+	"time"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/coreos/go-systemd/dbus"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/coreos/go-systemd/util"
+)
+
+func randomFreePort(t *testing.T) int {
+	l, err := net.Listen("tcp", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer l.Close()
+
+	addr := l.Addr().String()
+
+	n := strings.LastIndex(addr, ":")
+	port, err := strconv.Atoi(addr[n+1:])
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return port
+}
+
+func TestSocketActivation(t *testing.T) {
+	if !util.IsRunningSystemd() {
+		t.Skip("Systemd is not running on the host.")
+	}
+
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	port := randomFreePort(t)
+
+	echoImage := patchTestACI("rkt-inspect-echo.aci",
+		"--exec=/echo-socket-activated",
+		fmt.Sprintf("--ports=%d-tcp,protocol=tcp,port=%d,socketActivated=true", port, port))
+
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+
+	conn, err := dbus.New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rktTestingEchoService := `
+	[Unit]
+	Description=Socket-activated echo server
+
+	[Service]
+	ExecStart=%s
+	KillMode=process
+	`
+
+	rnd := r.Int()
+
+	cmd := fmt.Sprintf("%s --insecure-skip-verify run --mds-register=false %s", ctx.cmd(), echoImage)
+	serviceContent := fmt.Sprintf(rktTestingEchoService, cmd)
+	serviceTarget := fmt.Sprintf("rkt-testing-socket-activation-%d.service", rnd)
+
+	if err := ioutil.WriteFile(serviceTarget, []byte(serviceContent), 0666); err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(serviceTarget)
+
+	rktTestingEchoSocket := `
+	[Unit]
+	Description=Socket-activated netcat server socket
+
+	[Socket]
+	ListenStream=%d
+	`
+	socketContent := fmt.Sprintf(rktTestingEchoSocket, port)
+	socketTarget := fmt.Sprintf("rkt-testing-socket-activation-%d.socket", rnd)
+
+	if err := ioutil.WriteFile(socketTarget, []byte(socketContent), 0666); err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(socketTarget)
+
+	serviceTargetAbs, _ := filepath.Abs(serviceTarget)
```

Ignoring an error? Tsk, tsk, tsk.